plugins/modules/mso_schema_template_external_epg.py

```diff
@@ -260,11 +294,13 @@ def main():
             name=external_epg,
             displayName=display_name,
             vrfRef=vrf_ref,
+            extEpgType=external_epg_type,
             l3outRef=l3out_ref,
             preferredGroup=preferred_group,
         )
 
-        if anp is not None:
+        if external_epg_type == 'cloud':
+            del payload['l3outRef']
             payload['anpRef'] = anp_ref
```

Comment:
add else for the on-premises case to avoid the deletion of the key.

Reply:
if external_epg_type == 'cloud':
            payload['extEpgType'] = type
            payload['anpRef'] = anp_ref
else: